sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -4713,78 +4782,92 @@ SIREPO.app.directive('sbatchLoginModal', function() {
               </div>
             </div>
         `,
-        controller: function(requestSender, sbatchLoginStatusService, $element, $scope, $rootScope) {
-            $scope.otp = '';
-            $scope.password = '';
-            $scope.username = '';
-            var awaitingSendResponse = false;
-            var el = $('#sbatch-login-modal');
-            var errorResponse = null;
-            var onHidden = null;
-
-            el.on('hidden.bs.modal', function() {
-                var r = {'state': 'error', 'error': 'Please try again.'};
-                if (errorResponse) {
-                    r = {'state': 'error', 'error': errorResponse};
-                }
-                sbatchLoginStatusService.loginSuccess();
-                onHidden(r);
-                onHidden = null;
-                errorResponse = null;
-                $scope.otp = '';
-                $scope.password = '';
-                $scope.username = '';
-                $scope.sbatchLoginModalForm.$setPristine();
-                $scope.$apply();
-            });
-
-            function handleResponse(data) {
-                if (data.state === 'error') {
-                    errorResponse = data.error;
-                }
-                sbatchLoginStatusService.loggedIn = data.loginSuccess ? true: false;
-                $rootScope.$broadcast('sbatchLoginDone');
+        controller: function($element, $scope, $rootScope, authState, errorService, panelState, requestSender) {
+	    resetLoginFormTextFields();
```

Comment:
Seems like this should happen on hide. Otherwise, the credentials hang out in javascript. Can we catch hide and switch tabs?

Reply:
It needs to happen here too. For example, the invalid creds case. The modal is shown, a user submits invalid creds, we broadcast showSbatchLoginModal in the handling of the SRException, we come back here. The modal is never hidden in that workflow. 

I don't think we need to handle switch tabs specifically. When the modal is present it has to be hidden before a user is able to switch tabs.

Added reset on hidden 17a9330b3